### PR TITLE
MON-3895: Add tempo operator metrics

### PR DESCRIPTION
**Pull request by rubenvp8510**
* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml [hunk 1]
```diff
@@ -818,6 +818,26 @@ data:
...
     - '{__name__="jaeger_operator_instances_agent_strategies"}'
     #
+    # owners: (@tracing-team)
+    #
+    # Number of Tempo instances utilizing a specific backend storage type
+    - '{__name__="tempo_operator_tempostack_storage_backend"}'
+    #
+    # owners: (@tracing-team)
+    #
+    # Number of tempo instances managed or not by the tempo operator
```
Comment (simonpasquier) [inline manifests/0000_50_cluster-monitoring-operator_04-config.yaml:828]:
```suggestion
    # Number of Tempo instances per management state.
```

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml [hunk 2]
```diff
@@ -818,6 +818,26 @@ data:
...
+    - '{__name__="tempo_operator_tempostack_managed"}'
+    #
+    # owners: (@tracing-team)
+    #
+    # Number of tempo instances with multitenancy enabled or disable and which type of multitenancy are using.
+    - '{__name__="tempo_operator_tempostack_multi_tenancy"}'
+    #
+    # owners: (@tracing-team)
+    #
+    # Number of jaeger instances used certain agent strategy
```
Comment (simonpasquier) [inline manifests/0000_50_cluster-monitoring-operator_04-config.yaml:838]:
```suggestion
    # Number of Tempo stacks with Jaeger UI enabled/disabled.
```

**Review by simonpasquier — COMMENTED** [review]
For the record, the metrics are defined in https://github.com/grafana/tempo-operator/blob/main/internal/crdmetrics/tempo_stack.go

**Comment by rubenvp8510** [comment]
Hello @simonpasquier I renamed this to match the recording rules

Those are the rules defined in the tempo bundle:

```
apiVersion: monitoring.coreos.com/v1
kind: PrometheusRule
metadata:
  labels:
    app.kubernetes.io/managed-by: operator-lifecycle-manager
    app.kubernetes.io/name: tempo-operator
    app.kubernetes.io/part-of: tempo-operator
  name: tempo-operator-tempo-operator-prometheus-rules
spec:
  groups:
  - name: tempo-operator-monitoring.rules
    rules:
    - expr: sum by (type) (tempo_operator_tempostack_storage_backend)
      record: type:tempo_operator_tempostack_storage_backend:sum
    - expr: sum by (state) (tempo_operator_tempostack_managed)
      record: state:tempo_operator_tempostack_managed:sum
    - expr: sum by (type) (tempo_operator_tempostack_multi_tenancy)
      record: enabled:tempo_operator_tempostack_multi_tenancy:sum
    - expr: sum by (enabled) (tempo_operator_tempostack_jaeger_ui)
      record: enabled:tempo_operator_tempostack_jaeger_ui:sum
```

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml [hunk 3]
```diff
@@ -818,6 +818,26 @@ data:
     # Number of jaeger instances used certain agent strategy
     - '{__name__="jaeger_operator_instances_agent_strategies"}'
     #
+    # owners: (@tracing-team)
+    #
+    # Number of Tempo instances per backend storage type.
+    - '{__name__="type:tempo_operator_tempostack_storage_backend:sum"}'
```
Comment (simonpasquier) [inline manifests/0000_50_cluster-monitoring-operator_04-config.yaml:824]:
```suggestion
    - '{__name__="type:tempo_operator_tempostack_storage_backend:sum",type=~"azure|gcs|s3"}'
```

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml [hunk 4]
```diff
@@ -818,6 +818,26 @@ data:
...
+    #
+    # owners: (@tracing-team)
+    #
+    # Number of Tempo instances per management state.
+    - '{__name__="state:tempo_operator_tempostack_managed:sum"}'
```
Comment (simonpasquier) [inline manifests/0000_50_cluster-monitoring-operator_04-config.yaml:829]:
```suggestion
    - '{__name__="state:tempo_operator_tempostack_managed:sum",state=~"Managed|Unmanaged"}'
```

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml [hunk 5]
```diff
@@ -818,6 +818,26 @@ data:
...
+    #
+    # owners: (@tracing-team)
+    #
+    # Number of Tempo instances per multitenancy mode.
+    - '{__name__="type:tempo_operator_tempostack_multi_tenancy:sum"}'
```
Comment (simonpasquier) [inline manifests/0000_50_cluster-monitoring-operator_04-config.yaml:834]:
```suggestion
    - '{__name__="type:tempo_operator_tempostack_multi_tenancy:sum",type=~"enabled|disabled"}'
```

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml [hunk 6]
```diff
@@ -818,6 +818,26 @@ data:
...
+    #
+    # owners: (@tracing-team)
+    #
+    # Number of Tempo stacks with Jaeger UI enabled/disabled.
+    - '{__name__="enabled:tempo_operator_tempostack_jaeger_ui:sum"}'
```
Comment (simonpasquier) [inline manifests/0000_50_cluster-monitoring-operator_04-config.yaml:839]:
```suggestion
    - '{__name__="enabled:tempo_operator_tempostack_jaeger_ui:sum",enabled="true|false"}'
```

**Review by simonpasquier — COMMENTED** [review]
You'll to run "make generate" and commit all changes. I've added some suggestion to keep the cardinality within agreed bounds.

**Comment by rubenvp8510** [comment]
@simonpasquier I think this is ready

**Review by simonpasquier — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**